contributing.md

```diff
@@ -321,11 +321,17 @@ Example: `Fix typos in Foo scaladoc`
 `dev@spark.apache.org` first before proceeding to implement the change.
 
 <h3>Pull request</h3>
+Before creating a pull request in Apache Spark, it is important to check if tests can pass on your branch because 
+our GitHub Actions workflows automatically run tests for your pull request/following commits 
+and every run burdens the limited resources of GitHub Actions in Apache Spark repository.
+Below steps will take your through the process.
+
 
 1. <a href="https://help.github.com/articles/fork-a-repo/">Fork</a> the GitHub repository at 
 <a href="https://github.com/apache/spark">https://github.com/apache/spark</a> if you haven't already
-1. Clone your fork, create a new branch, push commits to the branch.
-1. Consider whether documentation or tests need to be added or updated as part of the change, 
+2. Go to "Actions" tab on your forked repository and enable "Build and test" and "Report test results" workflows  
```

Comment:
Just write `1.` on each of these and revert subsequent changes. The point is that markdown generates the correct numbering

Reply:
+1 for Sean's comment.

Reply:
Fixed